app/controllers/SummaryDeclarationController.scala

```diff
@@ -85,34 +85,66 @@ class SummaryDeclarationController @Inject() (val mcc: MessagesControllerCompone
         }
       } else ("", 0)
 
+      val schemeID = requestObject.getSchemeId
       val altAmendsActivity =
         getEntry[AltAmendsActivity](all, DataKey(ersUtil.ALT_AMENDS_ACTIVITY)).getOrElse(AltAmendsActivity(""))
-      val altActivity       = requestObject.getSchemeId match {
+      val altActivity       = schemeID match {
         case ersUtil.SCHEME_CSOP | ersUtil.SCHEME_SIP | ersUtil.SCHEME_SAYE => altAmendsActivity.altActivity
         case _ => ""
       }
-      Future(
-        Ok(
-          summaryView(
-            requestObject,
-            reportableEvents,
-            fileType,
-            fileNames,
-            fileCount,
-            groupScheme,
-            schemeOrganiser,
-            getCompDetails(all),
-            altActivity,
-            getAltAmends(all),
-            getTrustees(all)
+
+      if (validateCompanies(all, groupScheme) && validateAltAmends(all, altActivity, schemeID)) {
+        Future(
+          Ok(
+            summaryView(
+              requestObject,
+              reportableEvents,
+              fileType,
+              fileNames,
+              fileCount,
+              groupScheme,
+              schemeOrganiser,
+              getCompDetails(all),
+              altActivity,
+              getAltAmends(all),
+              getTrustees(all)
+            )
           )
         )
-      )
+      } else {
+        throw new Exception("Validation of companies or alt activities failed")
+      }
     } recover { case e: Throwable =>
       logger.error(s"[SummaryDeclarationController][showSummaryDeclarationPage] failed to load page with exception ${e.getMessage}.", e)
       getGlobalErrorPage
     }
 
+  private def validateCompanies(all: CacheItem, groupScheme: String): Boolean = {
+    val companiesExist = getCompDetails(all).companies.nonEmpty
+    if ((companiesExist && groupScheme == ersUtil.OPTION_YES) || (!companiesExist && groupScheme == ersUtil.OPTION_NO)) {
+      true
+    } else {
+      logger.error(s"[SummaryDeclarationController][showSummaryDeclarationPage] attempted to route to summary page with incompatible state: Companies Filled = $companiesExist. Group Scheme = $groupScheme")
+      false
+    }
+  }
+
+  def isValidAltActivity(altActivity: String, altAmendsEmpty: Boolean): Boolean =
```

Comment:
I think you can make this a bit simpler either:
```scala annotate
def isValidAltActivity(altActivity: String, altAmendsEmpty: Boolean): Boolean =
    (altAmendsEmpty && altActivity != ersUtil.OPTION_YES) ||
      (!altAmendsEmpty && altActivity == ersUtil.OPTION_YES)
```      
or
```scala annotate
def isValidAltActivity(altActivity: String, altAmendsEmpty: Boolean): Boolean =  altAmendsEmpty != (altActivity == ersUtil.OPTION_YES)
```

I think the first option is easier to read then the second but up to you.

Reply:
Cool, happy with this 👍